quilt/README.md

```diff
@@ -0,0 +1,241 @@
+# Running Ray with Quilt
+
+Quilt is a way to deploy and network containers, meaning that Quilt will boot
+machines and set up containers on supported cloud providers (AWS, GCE, Azure)
+for you. Learn more about this project [here](http://quilt.io).
+
+## Introduction
+Quilt uses a domain specific language, Stitch (files that end with `.spec`, to 
+specify distributed applications, independent of the specific infrastructure
+they run on. This folder contains [`ray.spec`](ray.spec) and 
+[`ray-runner.spec`](ray-runner.spec). `ray.spec` provides abstractions for
+creating ray containers - the function `ray.New` creates `nWorker` worker nodes,
+and one head node, all prefixed with `prefix`.
+
+`ray-runner.spec` defines the infrastructure When you run `ray-runner.spec`, Quilt will
+
+## Step 1: Installing Quilt
+
+### Install Go
+Quilt supports Go version 1.5 or later.
+
+Install Go using your package manager or on the 
+[Golang website](https://golang.org/doc/install).
+
+#### Set up GOPATH
+We recommend reading the overview to Go workplaces 
+[here](https://golang.org/doc/code.html).
+
+Before installing Quilt, you'll need to set up your `GOPATH`. Assuming the root
+of your Go workspace will be `~/gowork`, execute the following `export`
+commands in your terminal to set up your `GOPATH`:
+
+```bash
+export GOPATH=~/gowork
+export PATH=$PATH:$GOPATH/bin
+```
+
+It would be a good idea to add these commands to your `.bashrc` (or
+`.bash_profile` on OS X) so that they do not have to be run again.
+
+### Download and Install Quilt
+Clone the repository into your Go workplace: `go get github.com/NetSys/quilt`.
+
+This command also automatically installs Quilt. If the installation was
+successful, then the `quilt` command should execute in your shell.
+
+### QUILT_PATH
+Your `QUILT_PATH` will be where Quilt looks for imported specs, and where
+specs you download with `quilt get` get placed. You can set this to be anywhere,
+but by default, your `QUILT_PATH` is `~/.quilt`. To set a custom `QUILT_PATH`,
+follow the instructions
+[here](https://github.com/NetSys/quilt/blob/master/docs/Stitch.md#quilt_path).
+
+### Download Imported Specs
+If you look at [`ray.spec`](ray.spec), the first line imports 
+`github.com/NetSys/quilt/specs/stdlib/strings`. To resolve these imports, simply
+execute `quilt get github.com/amplab/ray`. This will clone the ray repository
```

Comment:
Update: I pulled the branch via `quilt get github.com/secant/ray/tree/quilt/quilt` which I would have expected to pull in any dependencies needed. But I still get the error:

```
ERROR [Aug 19 09:15:19.761] quilt/ray-runner.spec failed to compile. error=unable to open import github.com/NetSys/quilt/specs/stdlib/strings (path=/Users/johann/.quilt/github.com/NetSys/quilt/specs/stdlib/strings.spec)
```

I guess I need to understand the import system better. Also, it seems odd that I end up with whole git repos in my ~/.quilt directory.